src/entities/shapeEntity.js

```diff
@@ -1,58 +1,35 @@
-function ShapeEntity(Interceptor, shapeObject, shapeArgs, canvasX, canvasY) {
-  const self = this;
-  /* global BaseEntity */
-  BaseEntity.call(self, shapeObject, shapeArgs, canvasX, canvasY);
-  this.areaAbs = 0;
-  this.type = Interceptor.currentColor + ` ` + shapeObject.name;
-  this.area = 0;
-  this.length = 0;
-
-
-  this.populate = function(shapeObject, arguments, canvasX, canvasY) {
-    this.location = this.getLocation(shapeObject, arguments, canvasX, canvasY);
-    this.coordLoc = this.canvasLocator(shapeObject, arguments, canvasX, canvasY);
-    if(!shapeObject.name.localeCompare(`ellipse`) || !shapeObject.name.localeCompare(`rect`)  || !shapeObject.name.localeCompare(`triangle`) || !shapeObject.name.localeCompare(`quad`)) {
-      this.areaAbs = this.getObjectArea(shapeObject.name, arguments);
-      this.area = (this.getObjectArea(shapeObject.name, arguments) * 100 / (canvasX * canvasY)).toFixed(2) + `%`;
-    } else if(!shapeObject.name.localeCompare(`line`)) {
-      this.length = this.getLineLength(arguments);
-    }
+/* global BaseEntity */
+class ShapeEntity extends BaseEntity {
+  constructor(Interceptor, shapeObject, shapeArgs, canvasX, canvasY) {
+    super(Interceptor, shapeObject);
+    this.areaAbs = 0;
+    this.type = `${Interceptor.currentColor} ${shapeObject.name}`;
+    this.area = 0;
+    this.populate(shapeObject, shapeArgs, canvasX, canvasY);
   }
-
-  this.getAttributes = function() {
-    if((!shapeObject.name.localeCompare(`ellipse`)) || !shapeObject.name.localeCompare(`rect`)  || !shapeObject.name.localeCompare(`triangle`) || !shapeObject.name.localeCompare(`quad`)) {
-      return ({
-        type: this.type,
-        location: this.location,
-        coordinates: this.coordinates,
-        area: this.area
-      })
-    } 
-    else if(!shapeObject.name.localeCompare(`line`)) {
-      return ({
-        type: this.type,
-        location: this.location,
-        coordinates: this.coordinates,
-        length : this.length
-      })
-    } 
-    else {
-      return ({
-        type: this.type,
-        location: this.location,
-        coordinates: this.coordinates         
-      })
-    }
-  };
-
-  /* return length of lines */
```

Comment:
Looks like the special condition of lines was deleted during the merge

Reply:
Fixed in next commit